### PR TITLE
Update the  mongodb driver to 4.0

### PR DESCRIPTION
**Pull request by harisvsulaiman**
- [x] Fix type issues, 
- [x] Fix Tests

Resolves https://github.com/agenda/agenda/issues/1355

### CODE REVIEW

**Review by koresar — CHANGES_REQUESTED** [review]
Please update REAMDE and/or History with the upcoming SemVer major release or Agenda. It should sound something like:

> In order to support new MongoDB 5.0 and `mongodb` node.js driver/package the next release of Agenda will be major. The required node version will become >=12. The `mongodb` dependency version will become >=4.

(Please phrase it better than me, if you can. :) )

File: test/agenda.js [hunk 1]
```diff
@@ -851,7 +851,7 @@ describe("Agenda", () => {
 
   describe("process jobs", () => {
     // eslint-disable-line prefer-arrow-callback
-    it("should not cause unhandledRejection", async function () {
+    it.skip("should not cause unhandledRejection", async function () {
```
[the remark this answers is not in the document]
Comment (harisvsulaiman) [reply test/agenda.js]: Sorry my bad. forgot to remove the skip.

File: package.json [hunk 2]
```diff
@@ -53,12 +53,11 @@
     "debug": "~4.3.0",
     "human-interval": "~2.0.0",
     "moment-timezone": "~0.5.27",
-    "mongodb": "~3.6.6"
+    "mongodb": "^4.0.1"
```
[the remark this answers is not in the document]
Comment (harisvsulaiman) [reply package.json:56]: https://docs.mongodb.com/drivers/node/current/compatibility/ Shows compatibility unto `mongo 2.4` and `node v12.x.x`.

[the remark this answers is not in the document]
Comment (harisvsulaiman) [reply package.json:56]: This Compatibility will surely shrink once we introduce optional atomic operations via mongo transactions into agenda.

File: package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "dist"
   ],
   "engines": {
-    "node": ">=10.0.0"
+    "node": ">=12.9.0"
```
[the remark this answers is not in the document]
Comment (harisvsulaiman) [reply package.json:12]: Yes

[the remark this answers is not in the document]
Comment (koresar) [reply package.json:12]: Also, please merge latest master into your branch. There is yarn.lock conflict now.

**Comment by harisvsulaiman** [comment]
@koresar Merging to master beacuse tests run indefenitely due to #1369 . Will Fix after merge

**Pull request merged**